src/Roslyn.Suppressions/Roslyn.Suppressions/InProcess/SuppressedIssuesFileSynchronizer.cs

```diff
@@ -53,34 +57,59 @@ public IEnumerable<SonarQubeIssue> Get(string sonarProjectKey)
     [Export(typeof(ISuppressedIssuesFileSynchronizer))]
     internal sealed class SuppressedIssuesFileSynchronizer : ISuppressedIssuesFileSynchronizer
     {
+        private readonly IThreadHandling threadHandling;
         private readonly ISuppressedIssuesMonitor suppressedIssuesMonitor;
         private readonly ISonarQubeIssuesProvider suppressedIssuesProvider;
         private readonly ISuppressedIssuesFileStorage suppressedIssuesFileStorage;
         private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
 
         [ImportingConstructor]
-        public SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor, 
+        public SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor,
             ISonarQubeIssuesProvider suppressedIssuesProvider,
             ISuppressedIssuesFileStorage suppressedIssuesFileStorage,
             IActiveSolutionBoundTracker activeSolutionBoundTracker)
+            : this(suppressedIssuesMonitor,
+                suppressedIssuesProvider,
+                suppressedIssuesFileStorage,
+                activeSolutionBoundTracker,
+                new ThreadHandling())
+        {
+        }
+
+        internal SuppressedIssuesFileSynchronizer(ISuppressedIssuesMonitor suppressedIssuesMonitor,
+            ISonarQubeIssuesProvider suppressedIssuesProvider,
+            ISuppressedIssuesFileStorage suppressedIssuesFileStorage,
+            IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            IThreadHandling threadHandling)
         {
             this.suppressedIssuesMonitor = suppressedIssuesMonitor;
             this.suppressedIssuesProvider = suppressedIssuesProvider;
             this.suppressedIssuesFileStorage = suppressedIssuesFileStorage;
             this.activeSolutionBoundTracker = activeSolutionBoundTracker;
+            this.threadHandling = threadHandling;
 
             suppressedIssuesMonitor.SuppressionsUpdateRequested += SuppressedIssuesMonitor_SuppressionsUpdateRequested;
         }
 
         private void SuppressedIssuesMonitor_SuppressionsUpdateRequested(object sender, EventArgs e)
         {
-            var sonarProjectKey = activeSolutionBoundTracker.CurrentConfiguration.Project?.ProjectKey;
+            UpdateFileStorage();
+        }
 
-            if (!string.IsNullOrEmpty(sonarProjectKey))
+        public void UpdateFileStorage()
```

Comment:
Initially this code was being executed in the CTOR, however that caused the MEF export test to fail. ThreadHandling is not a MEF dependency so I could not inject a dummy mock to replace it and it was failing in the real implementation. So I moved the logic to a method and I'm explicitly calling this method in SonarLintIntegrationPackage.